### PR TITLE
DOCS-1795 - Sync the old docs file with the README

### PR DESCRIPTION
**Pull request by ddl-lgaddie**
### Link to JIRA

[DOCS-1795](https://dominodatalab.atlassian.net/browse/DOCS-1795)

### What issue does this pull request solve?

Currently the docs site shows this file for versions < 5.3:
https://github.com/cerebrotech/documentation/blob/main/content/api_guide/Python_wrapper_for_Domino_API.adoc

It's incomplete and out of date.

And for 5.3 and up it uses the `include` directive to pull in the Asciidoc-formatted version of the README from this repo.  This is the preferred method so that (a) docs can live close to the code and (b) we are single-sourcing instead of duplicating content and trying to keep it in sync.

### What is the solution?

This PR:

- Moves content from https://github.com/cerebrotech/documentation/blob/main/content/api_guide/Python_wrapper_for_Domino_API.adoc into https://github.com/dominodatalab/python-domino/blob/master/README.adoc (except where it was already present there).
- Syncs the content between `README.adoc` and `README.md` so that they're identical.
- Adds edits and formatting for clarity and conciseness.

To do:

- Do the same syncing/editing on the various release branches (such as `releases-1.2.0`).
- Determine which version of pydom shipped with each release of DAD/DSE and map those to Domino versions.
- Use conditional directives in https://github.com/cerebrotech/documentation/blob/main/content/api_guide/python-domino.adoc to point to the right version of the `README.adoc` file for each version of Domino.

### Testing

Briefly describe how the change was tested. The purpose of this section is that a reviewer can identify a test gap, if any.

_e.g. "I ran an upgrade from 4.2 to 4.6"._

- [ ] Unit test(s)

### Pull Request Reminders

- [ ] Has relevant documentation been updated?
- [ ] Does the code follow [Python Style Guide] (https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html)
- [ ] Update the [changelog](https://github.com/dominodatalab/python-domino/blob/master/CHANGELOG.md)
- [ ] Are the existing unit tests still passing?
- [ ] Have new unit tests been added to cover any changes to the code?
- [ ] Has the JIRA ticket(s) been linked above?

### CODE REVIEW

**Comment by ddl-lgaddie** [comment]
The items in the to-do list above are being handled in https://github.com/cerebrotech/documentation/pull/1069

**Comment by ddl-lgaddie** [comment]
@jp-domino I think we should move the compatibility matrix into the docs repo so that we can present a complete matrix:
https://github.com/cerebrotech/documentation/pull/1069/files#diff-de3f6d3326b3baeeaffac8f697b1429212051138163b945103d7274f04e34086R18

If you think it should also be in this repo's READMEs, then maybe we can move it to the top of the READMEs so that the docs repo's `include` directive can exclude it by starting one line below it.  For example, currently those directives https://github.com/cerebrotech/documentation/pull/1069/files#diff-de3f6d3326b3baeeaffac8f697b1429212051138163b945103d7274f04e34086R16[start at line 2](url) to exclude the README's title in favor of the Asciidoc file's title.

File: README.adoc [hunk 1]
```diff
@@ -1,11 +1,16 @@
 = python-domino
 
-This library provides bindings for the Domino APIs.  See this documentation for details about the APIs:
+:latest-version: 1.2.0
```
Comment (ddl-lgaddie) [inline README.adoc:3]:
```suggestion
:latest-version: 1.2.1
```

File: README.md [hunk 2]
```diff
@@ -261,9 +267,9 @@ Get `stdout` emitted by a particular execution.
 
 See these code example files:
 
--   [`upload_file.py`](https://github.com/dominodatalab/python-domino/blob/release-1.1.0/examples/upload_file.py)
+-   [`upload_file.py`](https://github.com/dominodatalab/python-domino/blob/release-1.2.0/examples/upload_file.py)
```
Comment (ddl-olsonJD) [inline README.md:270]: Do we not want to use the latest version for these 1.2.1?

File: README.md [hunk 3]
```diff
@@ -175,9 +181,9 @@ Remove a tag from a project.
 
 See these code example files:
 
--   [`start_run_and_check_status.py`](https://github.com/dominodatalab/python-domino/blob/release-1.1.0/examples/start_run_and_check_status.py)
+-   [`start_run_and_check_status.py`](https://github.com/dominodatalab/python-domino/blob/release-1.2.0/examples/start_run_and_check_status.py)
 
--   [`export_runs.py`](https://github.com/dominodatalab/python-domino/blob/release-1.1.0/examples/export_runs.py)
+-   [`export_runs.py`](https://github.com/dominodatalab/python-domino/blob/release-1.2.0/examples/export_runs.py)
```
Comment (ddl-olsonJD) [inline README.md:186]: can we use a consistent version number, unless needed to be specified?

**Review by ddl-olsonJD — CHANGES_REQUESTED** [review]

**Review by ddl-olsonJD — APPROVED** [review]

**Pull request merged**